src/components/PostDetails.tsx

```diff
@@ -3,59 +3,35 @@ import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
 import * as commentsApi from '../api/comments';
-
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  addNewComment,
+  changeError,
+  fetchComments,
+  removeComment,
+} from '../features/commentsSlice';
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-  const [visible, setVisible] = useState(false);
-
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
+  const dispatch = useAppDispatch();
+  const {
+    comments,
+    loading: loaded,
+    error: hasError,
+  } = useAppSelector(state => state.comments);
 
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
-    setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
-    }
-  };
+  const [visible, setVisible] = useState(false);
 
   useEffect(() => {
-    loadComments();
-  }, []);
+    setVisible(false);
 
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+    dispatch(fetchComments(post.id));
```

Comment:
The 'fetchComments' action is dispatched with 'post.id' as an argument. Ensure that 'fetchComments' is designed to accept and handle this parameter correctly.